test/e2e/instrumentation/events.go

```diff
@@ -0,0 +1,182 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package instrumentation
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	v1beta1 "k8s.io/api/events/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	eventsv1beta1 "k8s.io/client-go/kubernetes/typed/events/v1beta1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/instrumentation/common"
+
+	"github.com/onsi/ginkgo"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	eventRetryPeriod  = 1 * time.Second
+	eventRetryTimeout = 1 * time.Minute
+)
+
+func testEvent(name, label string) *v1beta1.Event {
+	someTime := metav1.MicroTime{Time: time.Unix(1505828956, 0)}
+	return &v1beta1.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				label: "true",
+			},
+		},
+		EventTime:           someTime,
+		Note:                "This is " + name,
+		Reason:              "Test",
+		Type:                "Normal",
+		ReportingController: "test-controller",
+		ReportingInstance:   "test-node",
+	}
+}
+
+func listEvents(name, namespace string, client eventsv1beta1.EventInterface) bool {
+	eventsList, err := client.List(context.TODO(), metav1.ListOptions{
+		LabelSelector: "testevent-constant=true",
+	})
+	framework.ExpectNoError(err, "failed list all events")
+
+	foundCreatedEvent := false
+	for _, val := range eventsList.Items {
+		if val.ObjectMeta.Name == name && val.ObjectMeta.Namespace == namespace {
+			foundCreatedEvent = true
+			break
+		}
+	}
+	return foundCreatedEvent
+}
+
+var _ = common.SIGDescribe("Events API", func() {
+	f := framework.NewDefaultFramework("events")
+	client := f.ClientSet.EventsV1beta1().Events(f.Namespace.Name)
+	clientAllNamespaces := f.ClientSet.EventsV1beta1().Events(metav1.NamespaceAll)
+
+	/*
+		Release : v1.19
+		Testname: Event resource lifecycle
+		Description: Create an event, the event MUST exist.
+		The event is patched with a new note, the check MUST have the update note.
+		The event is deleted and MUST NOT show up when listing all events.
+	*/
+	ginkgo.It("should ensure that an event can be fetched, patched, deleted, and listed", func() {
+		eventTestName := "event-test"
+
+		ginkgo.By("creating a test event")
+		// create a test event in test namespace
+		_, err := client.Create(context.TODO(), testEvent(eventTestName, "testevent-constant"), metav1.CreateOptions{})
+		framework.ExpectNoError(err, "failed to create test event")
+
+		ginkgo.By("listing all events in all namespaces")
+		// get a list of Events in all namespaces to ensure endpoint coverage
```

Comment:
same here (and probably in other places too)

Reply:
Yeah - it seems all the comments are useless.

Reply:
Done.